File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
class Solution {
    public int solution(String word) {
        String vowel = "AEIOU";
        int[] numbers = {781, 156, 31, 6, 1};
        int answer = 0;
        
        for(int i = 0; i < word.length(); i++) {
            char c = word.charAt(i);
            int index = 0;
            for(int j = 0; j < 5; j++) {
                if(c == vowel.charAt(j)) {
                    index = j;
                    break;
                }
            }
            answer += index * numbers[i] + 1;
        }
        
        
        return answer;
    }
}

File: Python3/프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    vowels = ['A', 'E', 'I', 'O', 'U']
    counts = [781, 156, 31, 6, 1]    
    answer = 0
    
    for i, value in enumerate(word):
        idx = vowels.index(value)
        answer += idx * counts[i] + 1
    return answer